xds/src/main/java/io/grpc/xds/RingHashLoadBalancer.java

```diff
@@ -334,23 +339,30 @@ public static EquivalentAddressGroup stripAttrs(EquivalentAddressGroup eag) {
   }
 
   private static final class RingHashPicker extends SubchannelPicker {
+    private final Random random = new Random();
     private final SynchronizationContext syncContext;
     private final List<RingEntry> ring;
     // Avoid synchronization between pickSubchannel and subchannel's connectivity state change,
     // freeze picker's view of subchannel's connectivity state.
     // TODO(chengyuanzhang): can be more performance-friendly with
     //  IdentityHashMap<Subchannel, ConnectivityStateInfo> and RingEntry contains Subchannel.
     private final Map<Endpoint, SubchannelView> pickableSubchannels;  // read-only
+    private final String requestHashHeader;
```

Comment:
I think we want to move the construction of `Metadata.Key<String>` earlier and store the object here. It'd be `null` when the hash header is empty string. We should also store the key in `RingHashLoadBalancer.requestHashHeader`. It's up to you whether we store it in RingHashConfig.

Reply:
Done.
* Moved construction of `Metadata.Key<String>` earlier to `acceptResolvedAddresses()` and store the object in both `RingHashPicker.requestHashHeaderKey` and `RingHashLoadBalancer.requestHashHeaderKey`.
* Added a `@Nullable` annotation as the key could be `null` if the request hash header is the empty string.
* I decided to leave the `RingHashConfig` as is to keep the intent focused and data type primitive.
* Updated name from `requestHashHeader` to `requestHashHeaderKey` to better reflect that this is a metadata key for the request hash header.